internal/auth/providers/singleflight_middleware.go

```diff
@@ -124,7 +126,14 @@ func (p *SingleFlightProvider) ValidateGroupMembership(email string, allowedGrou
 	sort.Strings(allowedGroups)
 	response, err := p.do("ValidateGroupMembership", fmt.Sprintf("%s:%s", email, strings.Join(allowedGroups, ",")),
 		func() (interface{}, error) {
-			return p.provider.ValidateGroupMembership(email, allowedGroups, accessToken)
+			switch v := p.provider.(type) {
```

Comment:
hmm this is really not what I anticipated us doing. the `GroupsCache` should be fulfilling the entire interface, instead of just the one method.